src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -146,7 +152,8 @@ public void testFromXContent() throws IOException {
     @Test
     public void testToQuery() throws IOException {
         QueryParseContext context = new QueryParseContext(index, queryParserService);
-        assertLuceneQuery(this.testQuery, this.testQuery.toQuery(context));
+        context.setAllowUnmappedFields(true);
```

Comment:
Why are we using randomly generated field names? Can we instead set up mappings at the beginning of the test and select from the fields generated there? I don't think we should be testing *all* of our query tests through unmapped fields; this is a corner case. We should, however, have some tests to check the behavior of unmapped fields with those settings, but the rest of the tests should operate on valid fields.